### DISCUSSION TITLE
Question regarding Set commands

### DISCUSSION THREAD

**Discussion by joerghohn** [jblance/mpp-solar#86] [Q&A]
Hi, 

tried to set the backlights or buzzer on/off (before starting to set something else)
According to command list this should be the PEPD commands and accoring to PI30 Protocol Description the backlight should be X

So in this case my assumption is I set the baclights 
on: mpp-solar -p /dev/ttyUSB0 -P PI30 -c PEX 
of: mpp-solar -p /dev/ttyUSB0 -P PI30 -c PDX

Is that right?

Jörg

**Comment by jblance** [comment]
That is my understanding (though I have never tried this command either)
Does it not work?

**Reply by jblance** [reply]
Actually looking at the command definition, I dont think it will work as coded, its a bug in my code

**Reply by joerghohn** [reply]
Thanks for the quick answers.

**Reply by joerghohn** [reply]
You are right, it does not work. All the PE/PD commands show an "error NAK"

**Reply by jblance** [reply]
fixed the code to allow un-defined commands
I cant seem to find a version of PE/PD that works? do you have any idea of what command should work?

**Reply by joerghohn** [reply]
With WatchPower (the software that was delivert with the inverter) I have en-/disabled Backlight and Buzzer, I did not test anything else but this 2 worked. I can not say if this is also using the PE/PD commands for this.

**Reply by jblance** [reply]
Could you do a wireshark track of the Enable disable?

**Reply by jblance** [reply]
seems to work with lowercase characters
```
mpp-solar -c PEa
Command: PEa - Set the enabled state of various Inverter settings (e.g. buzzer, overload, interrupt alarm)
------------------------------------------------------------
Parameter                     	Value           Unit
pe                            	ACK     
```

**Comment by SunnyPerth** [comment]
Hi Jblance, 

first, great piece of code here - love it. Quick question on setting values - similar topic as above, so post question here. 
I'm trying to change the AC charge current, but not sure how I pass on the **value** to the command. I've tried just adding the value behind it, or with a space and various other ways. Any idea? 

Cheers
y. 




```
root@raspberrypi:~# mpp-solar -p /dev/hidraw0 -P PI30 -c MUCHGC
Parameter                     	Value           Unit
error                         	NAK            	    
root@raspberrypi:~# mpp-solar -p /dev/hidraw0 -P PI30 -c MUCHGC 30
usage: mpp-solar [-h] [-n NAME] [-p PORT] [--porttype PORTTYPE]
                 [-P {PI00,PI16,PI17,PI18,PI30,PI30REVO,PI41,VED,DALY,DALY40}]
                 [-T TAG] [-b BAUD] [-o [OUTPUT]] [--keepcase]
                 [--filter FILTER] [--exclfilter EXCLFILTER] [-q MQTTBROKER]
                 [--mqttport MQTTPORT] [--mqtttopic MQTTTOPIC]
                 [--mqttuser MQTTUSER] [--mqttpass MQTTPASS] [-c [COMMAND]]
                 [-C [CONFIGFILE]] [--daemon] [--getstatus] [--getsettings]
                 [-v] [-D] [-I]
mpp-solar: error: unrecognized arguments: 30
root@raspberrypi:~# mpp-solar -p /dev/hidraw0 -P PI30 -c MUCHGC30
Command: MUCHGC30 - No description found
------------------------------------------------------------
Parameter                     	Value           Unit
warning                       	No definition for command MUCHGC30 in protocol b'PI30'	    
                              	(ACK9 
root@raspberrypi:~# 
```

**Reply by jblance** [reply]
I think it needs 3 digits can you try `mpp-solar -p /dev/hidraw0 -P PI30 -c MUCHGC030`
also `mpp-solar -p /dev/hidraw0 -P PI30 -c QMUCHGCR` should give allowed options

**Reply by SunnyPerth** [reply]
That's it. Genius. Thanks Mate. (I had the Q in front of the comment, but that's for the query I realise now).
